File: leetcode/2097-watering-plants/wateringPlants.ts
function wateringPlants(plants: number[], capacity: number): number {
  let steps = 0;
  let rest = capacity;

  for (const [index, plant] of plants.entries()) {
    if (rest >= plant) {
      rest = rest - plant;
      steps = steps + 1;
    } else {
      rest = capacity - plant;
      steps = steps + 2 * index + 1
    }
  }


  return steps;
};

export default wateringPlants

File: leetcode/2097-watering-plants/wateringPlants.go
package leetcode

func wateringPlants(plants []int, capacity int) int {
	var steps = 0
	var rest = capacity

	for index, plant := range plants {
		if rest >= plant {
			steps = steps + 1
			rest = rest - plant
		} else {
			steps = steps + 2*index + 1
			rest = capacity - plant
		}
	}

	return steps
}
